lib/features/sign_in/sign_in.dart

```diff
@@ -1,12 +1,98 @@
 import 'package:flutter/material.dart';
+import 'package:qadam_education/constants/app_color.dart';
 
-class SignIn extends StatelessWidget {
+import 'input_field.dart';
+
+class SignIn extends StatefulWidget {
   const SignIn({super.key});
 
+  @override
+  State<SignIn> createState() => _SignInState();
+}
+
+class _SignInState extends State<SignIn> {
+  late Size mediaSize;
+  TextEditingController emailController = TextEditingController();
+  TextEditingController passwordController = TextEditingController();
+
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('Sign in page'),
+    mediaSize = MediaQuery.of(context).size;
```

Comment:
no need to use global field
just can use it locally final width = MediaQuery.of(context).size.width

Reply:
MediaQuery.sizeOf(context) is faster that MediaQuery.of(context).size